src/content/profile-data.js

```diff
@@ -122,53 +126,128 @@ export function defaultThreadOrder(threads: Thread[]) {
+    // index. Since Maps can't be keyed off of two values, do a little math to key off
+    // of both values: newStackPrefix * potentialFrameCount + frame => newStackIndex
+    const prefixStackAndFrameToStack = new Map();
+    const potentialFrameCount = newFrameTable.length * 2;
     const oldStackToNewStack = new Map();
-    const frameCount = frameTable.length;
-    const prefixStackAndFrameToStack = new Map(); // prefixNewStack * frameCount + frame => newStackIndex
 
-    function convertStack(stackIndex) {
-      if (stackIndex === null) {
+    function convertStack(oldStack) {
+      if (oldStack === null) {
         return null;
       }
-      let newStack = oldStackToNewStack.get(stackIndex);
+      let newStack = oldStackToNewStack.get(oldStack);
       if (newStack === undefined) {
-        const prefixNewStack = convertStack(stackTable.prefix[stackIndex]);
-        const frameIndex = stackTable.frame[stackIndex];
-        const funcIndex = frameTable.func[frameIndex];
-        if (!funcTable.isJS[funcIndex]) {
-          newStack = prefixNewStack;
-        } else {
-          const prefixStackAndFrameIndex = (prefixNewStack === null ? -1 : prefixNewStack) * frameCount + frameIndex;
+        // No stack was found, generate a new one.
+        const oldStackPrefix = stackTable.prefix[oldStack];
+        const newStackPrefix = convertStack(oldStackPrefix);
+        const frameIndex = stackTable.frame[oldStack];
+        const funcIndex = newFrameTable.func[frameIndex];
+        const oldStackIsPlatform = !newFuncTable.isJS[funcIndex];
+        let keepStackFrame = true;
+
+        if (oldStackIsPlatform) {
+          if (oldStackPrefix !== null) {
+            // Only keep the platform stack frame if the prefix is JS.
+            const prefixFrameIndex = stackTable.frame[oldStackPrefix];
+            const prefixFuncIndex = newFrameTable.func[prefixFrameIndex];
+            keepStackFrame = newFuncTable.isJS[prefixFuncIndex];
+          }
+        }
+
+        if (keepStackFrame) {
+          // Convert the old JS stack to a new JS stack.
+          const prefixStackAndFrameIndex = (newStackPrefix === null ? -1 : newStackPrefix) * potentialFrameCount + frameIndex;
           newStack = prefixStackAndFrameToStack.get(prefixStackAndFrameIndex);
           if (newStack === undefined) {
             newStack = newStackTable.length++;
-            newStackTable.prefix[newStack] = prefixNewStack;
-            newStackTable.frame[newStack] = frameIndex;
+            newStackTable.prefix[newStack] = newStackPrefix;
+            if (oldStackIsPlatform) {
+              // Create a new platform frame
+              const newFuncIndex = newFuncTable.length++;
+              newFuncTable.name.push(stringTable.indexForString('Platform'));
+              newFuncTable.resource.push(null);
+              newFuncTable.address.push(stringTable.indexForString(''));
```

Comment:
The address of a func is not an index into the string table, it's an offset from the library's base address. It's only meaningful for C++ functions.